modules/dnn/include/opencv2/dnn/dict.hpp

```diff
@@ -97,6 +103,7 @@ struct CV_EXPORTS_W DictValue
 private:
 
     Param type;
+    int ndims;
```

Comment:
We should not break design of classic [sum type (variant)](https://en.wikipedia.org/wiki/Tagged_union) implementation.
Only 2 fields are expected:
- tag of type
- union of stored types

---

`ndims` and other things must go to underlying (new) types instead.
Also would not see such massive and incomplete changes for already existed code.

Reply:
Good point. Fixed. Instead of storing ndims separately, it's now a part of `std::pair<Mat, int>`.